src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs

```diff
@@ -223,133 +202,104 @@ public static SBMessage AmqpMessageToSBMessage(AmqpMessage amqpMessage)
                         }
                         dataSegments.AddRange(arraySegmentValue);
                     }
-                }
                 sbMessage = new SBMessage(dataSegments.ToArray());
             }
             else
             {
                 sbMessage = new SBMessage();
             }
 
-            SectionFlag sections = amqpMessage.Sections;
+            var sections = amqpMessage.Sections;
```

Comment:
Suggestion: I'm a big advocate of `var`. Saying that, I think it would benefit from naming the variables to hint at what they stand for. For me as an inexperienced in AMQP, sections or section flags doesn't say a lot. But being able to look it up by the proper name (Section Flags), I'd be able to understand better. My personal rule of thumb - name var-ed variabled as close to what the represent so that actual type won't be even needed 🙂